Update for pgen v0.7

0xc0170 opened the pull request:
I would like to keep this open for a while, for people to test. I'll have few more patches in the queue for pgen v0.6 anyway, which I'll test with valinor.

There's now debugger setting, which I set to cmsis-dap by default. I have one target - silabs giant gecko which has j-link on board. How can we propagate this info to pgen to set jlink in the exported project file?

```
'debugger' : 'cmsis-dap',   # TODO: find out what debugger is connected
```

0xc0170 commented:
One more note, I haven't updated tests yet, another TODO here.

0xc0170 commented:
I am in process of finishing pgen 0.7, and updating this PR to 0.7 directly as it simplifies things. I'll fix tests shortly.

*in progress*

0xc0170 commented:
I fixed locally tests and also pgen to provide all valinor needs. Once pgen 0.7 is out, tests should get the new version and this should become green.

@autopulated please review

0xc0170 commented on setup.py:
```diff
@@ -42,7 +42,7 @@ def read(fname):
         'colorama>=0.3,<0.4',
         'pyOCD>=0.3,<1.0',
-        'project_generator>=0.5.7,<1.0'
+        'project_generator>=0.7.0,<=0.8.0'
```
We shall use just 0.7.x, as 0.8 might have some breaking change

0xc0170 commented on valinor/main.py:
```diff
@@ -98,46 +101,46 @@ def main():
     # renamed to have the .axf extension)
-    executable = tool.fixup_executable(exporter, args.executable, ide_tool)
+    executable = args.executable
+    if ide_tool == 'uvision':
```
I removed fix up executable as it did not make sense to have it in Project and Tools, it's just needed for uvision so I placed it here.

autopulated replied:
OK, makes sense.

autopulated commented:
Looks good – I don't have the setup to test locally right now, but if you've tested uvision and GDB then I'll go ahead and merge?

0xc0170 commented:
I haven't tested gdb yet, and lets wait until 0.7 makes it to pypi (I'll aim for tomorrow)

autopulated commented:
ok

autopulated commented:
just checking – 0.7 doesn't seem to be in PyPI yet?

0xc0170 commented:
There were few bugs discovered during testing which needs fixing. I'll notify you once 0.7 is out.

autopulated commented:
ack

matthewelse commented:
There was an issue I came across while trying `yt debug`, which I've fixed now (I'll send a pr to pgen in a minute). I'm going to see if I can find any more bugs which propagate through valinor to yotta...

0xc0170 commented:
I rebased this, fixed some outstanding issues. I tested gdb and uvision for k64f . The only thing holding this is v0.7 to be released. I estimate it will go out this week :crossed_flags:

0xc0170 commented:
@autopulated pgen v0.7 is out, tests are green. Give it a try

autopulated commented:
:tada: I'll test and merge next time I'm near some hardware.

autopulated commented:
:+1:

0xc0170 commented:
:dango:
